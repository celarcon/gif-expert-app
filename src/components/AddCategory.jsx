import { useState } from "react"
import PropTypes from "prop-types"

export const AddCategory = ({ setCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (e) => {
        e.preventDefault()
        setCategory(inputValue)
        setInputValue('')
    }
    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text"
                placeholder="Buscar"
                value={inputValue}
                onChange={handleInputChange}/>
        </form>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}