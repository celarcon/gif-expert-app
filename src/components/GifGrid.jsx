import { GifItem } from "../components/GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"
import  SvgComponent  from "../assets/spinner"

export const GifGrid = ({ category }) => {

    const { images, isloading } = useFetchGifs(category)

    return(
            <div className="card-grid">
                { isloading && <SvgComponent /> }
                <h2>{category}</h2>
                { images.map((image) =>  <GifItem key={image.id} {...image} />)}
            </div>
    )
}