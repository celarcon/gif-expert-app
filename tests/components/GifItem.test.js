import { render } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en <GiftItem />', () => {

    const title = 'Titulo de prueba'
    const url = 'https://media1.giphy.com/media/l0Hl6qQ7zQkIkIp2Q/200w.gif'
    test('Debe hacer match con el enapshot', () => {
        const { container } = render ( <GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    })
})