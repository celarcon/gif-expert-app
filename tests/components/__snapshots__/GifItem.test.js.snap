// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe hacer match con el enapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Titulo de prueba
    </p>
    <img
      alt="Titulo de prueba"
      src="https://media1.giphy.com/media/l0Hl6qQ7zQkIkIp2Q/200w.gif"
    />
  </div>
</div>
`;
